### ISSUE TITLE
Bump version of pkgdown

### ISSUE THREAD

**Issue by StrikerRUS** [microsoft/LightGBM#3036]
## Summary

Build R documents with new version of `pkgdown`.

## Motivation

`1.5` version introduces `repo$url` config option. This option will allow to drop our ugly postprocessing JS workaround for incorrect source files links as our R package is located in a subdirectory:
https://github.com/microsoft/LightGBM/blob/master/R-package/pkgdown/extra.js

## References
https://github.com/microsoft/LightGBM/blob/2c18a0f3ed732df77d4304ed1d2b46a345d661c8/docs/conf.py#L238

> You can now control the links to source files (in reference pages and
articles) and issues and users (in the NEWS) with new repo$url config
option (https://github.com/r-lib/pkgdown/issues/1238). This makes it easier to use pkgdown with GitHub enterprise, packages in subdirectories, and other source hosts (like bitbucket).

https://github.com/r-lib/pkgdown/releases/tag/v1.5.0

**Comment by StrikerRUS** [comment]
Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.

**Comment by jameslamb** [comment]
re-opening this since I think it will be fixed by open PR #3072 

Leaving some notes for myself. You can see the details of the relevant `pkgdown` changes in `R/build.r` section of https://github.com/r-lib/pkgdown/pull/1253/files, namely:

```text
#' @section YAML config - repo:
#' pkgdown automatically generates links to the source repository in a few
#' places
#'
#' * Articles and documentation topics are linked back to the
#'   underlying source file.
#'
#' * The NEWS automatically links issue numbers and user names.
#'
#' * The homepage provides a link to  "Browse source code"
#'
#' pkgdown automatically figures out the necessary URLs if you link to a GitHub
#' or GitLab repo in your `BugReports` or `URL` field. Otherwise, you can
#' supply your own in the `repo` component:
#'
#' ```yaml
#' repo:
#'   url:
#'     home: https://github.com/r-lib/pkgdown/
#'     source: https://github.com/r-lib/pkgdown/blob/master/
#'     issue: https://github.com/r-lib/pkgdown/issues/
#'     user: https://github.com/
#' ```
#'
#' * `home`: path to package home on source code repository.
#' * `source:`: path to source of individual file in master branch.
#' * `issue`: path to individual issue.
#' * `user`: path to user.
#'
#' The varying components (e.g. path, issue number, user name) are pasted on
#' the end of these URLs so they should have trailing `/`s.
```

So to check that a fix  to this worked, after removing  `R-package/pkgdown/extra.js`, check that:

* GitHub button in the top right corner of the site should point to https://github.com/microsoft/LightGBM/tree/master/R-package
* `Source:` links  on each function's page should point to the right file in GitHub, like this:

![image](https://user-images.githubusercontent.com/7608904/81633727-03d69b00-93d3-11ea-805e-76fd19e65d8f.png)
